l10n_jp_summary_invoice_sale_partner/models/account_move.py

```diff
@@ -0,0 +1,16 @@
+# Copyright 2025 Quartile (https://www.quartile.co)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    def get_summary_invoice_sale_partner(self):
+        partner_type = self.company_id.summary_invoice_sale_partner_type
+        if not partner_type:
+            return False
+        if partner_type == "partner":
+            return self.sale_partner_id
+        return self.partner_shipping_id
```

Comment:
There could be cases where both sold-to and ship-to partners should show on the document. Having separate config fields such as `show_sale_partner_in_summary_invoice_line_group` and `show_shipping_partner_in_summary_invoice_line_group` would make a better design IMO.